jscomp/syntax/src/res_printer.ml

```diff
@@ -2416,6 +2416,25 @@ and print_pattern ~state (p : Parsetree.pattern) cmt_tbl =
       Doc.concat [Doc.text "..."; print_ident_path ident cmt_tbl]
     | Ppat_type ident ->
       Doc.concat [Doc.text "#..."; print_ident_path ident cmt_tbl]
+    | Ppat_record (rows, _)
+      when ParsetreeViewer.has_dict_pattern_attribute p.ppat_attributes ->
+      Doc.concat
+        [
+          Doc.text "dict{";
+          Doc.indent
+            (Doc.concat
```

Comment:
Is all this printing logic specific to dicts, or is it partially shared with something else.
It's pretty short anyway so it looks good.